compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBranchesScreen.kt

```diff
@@ -0,0 +1,179 @@
+package com.bottlerocketstudios.compose.repository
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalTextStyle
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import com.bottlerocketstudios.compose.R
+import com.bottlerocketstudios.compose.resources.Colors
+import com.bottlerocketstudios.compose.resources.Dimens
+import com.bottlerocketstudios.compose.resources.brown_grey
+import com.bottlerocketstudios.compose.resources.transparent
+import com.bottlerocketstudios.compose.util.Preview
+import com.bottlerocketstudios.compose.util.PreviewAll
+import com.bottlerocketstudios.compose.util.formattedUpdateTime
+import com.bottlerocketstudios.launchpad.compose.bold
+import java.time.Clock
+import java.time.ZonedDateTime
+
+@Composable
+fun RepositoryBranchesScreen(state: RepositoryBranchesScreenState) {
+    val searchState = remember {
+        mutableStateOf(TextFieldValue())
+    }
+    LazyColumn(
+        modifier = Modifier.fillMaxSize()
+    ) {
+        item {
+            HeaderItem(
+                path = state.path.value,
+                itemCount = state.itemCount.value
+            )
+        }
+        item {
+            SearchView(state = searchState, placeHolder = stringResource(id = R.string.home_branches_search))
+        }
+        items(items = state.branchItems.value.filter { it.name.contains(searchState.value.text) }) {
+            BranchItem(
+                item = it
+            )
+        }
```

Comment:
@LuisRocketRangel Thanks for adding the Search view! This is a nice addition :)

Reply:
Woo!